File: scripts/repro/mock_llm.py
from fastapi import FastAPI
import uvicorn
from pydantic import BaseModel
import sys
import time

app = FastAPI()

class Message(BaseModel):
    role: str
    content: str

class ChatRequest(BaseModel):
    model: str
    messages: list[Message]
    max_tokens: int | None = None

@app.post("/v1/chat/completions")
async def chat_completions(req: ChatRequest):
    print(f"[MOCK LLM] Received request for model: {req.model}", file=sys.stderr)
    
    tier = "Unknown"
    node = "Unknown"
    
    if "brain" in req.model.lower():
        tier = "Tier 1 (Instant)"
        node = "Node B (Director)"
    elif "qwen" in req.model.lower():
        tier = "Tier 2 (Strategic)"
        node = "Node D (Core)"

    response_content = f"<think>\nProcessing route via {node}...\n</think>\nMesh Dry Run Acknowledged. I am {node} operating at {tier}."
    
    return {
        "id": "chatcmpl-mock123",
        "object": "chat.completion",
        "created": int(time.time()),
        "model": req.model,
        "choices": [{
            "index": 0,
            "message": {
                "role": "assistant",
                "content": response_content
            },
            "finish_reason": "stop"
        }],
        "usage": {"prompt_tokens": 10, "completion_tokens": 20, "total_tokens": 30}
    }

@app.get("/health")
async def health():
    return {"status": "ok"}

if __name__ == "__main__":
    port = int(sys.argv[1]) if len(sys.argv) > 1 else 8080
    print(f"[MOCK LLM] Starting mock server on port {port}", file=sys.stderr)
    uvicorn.run(app, host="0.0.0.0", port=port, log_level="error")

File: scripts/repro/mock_llm.js
const http = require('http');
const port = 8080;

const server = http.createServer((req, res) => {
  let body = '';
  req.on('data', chunk => { body += chunk.toString(); });
  req.on('end', () => {
    console.log(`[MOCK LLM] ${req.method} ${req.url}`);
    if (req.method === 'POST' && req.url === '/v1/chat/completions') {
      try {
        const data = JSON.parse(body || '{}');
        const model = data.model || 'unknown';
        let tier = "Unknown";
        let node = "Unknown";
        
        if (model.toLowerCase().includes("carnice")) { 
          tier = "Tier 1 (Instant)"; 
          node = "Node B (Director)"; 
        } else if (model.toLowerCase().includes("qwen")) { 
          tier = "Tier 2 (Strategic)"; 
          node = "Node D (Core)"; 
        }

        res.writeHead(200, { 'Content-Type': 'application/json' });
        res.end(JSON.stringify({
          id: "chatcmpl-mock123",
          object: "chat.completion",
          created: Math.floor(Date.now() / 1000),
          model: model,
          choices: [{
            index: 0,
            message: { 
              role: "assistant", 
              content: `<think>\nProcessing route via ${node}...\n</think>\nMesh Dry Run Acknowledged. I am ${node} operating at ${tier}.` 
            },
            finish_reason: "stop"
          }],
          usage: { prompt_tokens: 10, completion_tokens: 20, total_tokens: 30 }
        }));
      } catch (e) {
        res.writeHead(400);
        res.end("Bad Request");
      }
    } else if (req.url === '/health') {
      res.writeHead(200, { 'Content-Type': 'application/json' }); 
      res.end(JSON.stringify({ status: "ok" }));
    } else {
      res.writeHead(404); 
      res.end("Not Found");
    }
  });
});

server.listen(port, '0.0.0.0', () => console.log(`Mock LLM listening on 0.0.0.0:${port}`));
